crates/chain/tests/multi_node/mempool_tests.rs

```diff
@@ -217,6 +225,1117 @@ async fn mempool_persistence_test() -> eyre::Result<()> {
+    b_node.upload_chunks(&a_blk1_tx1).await?;
+    b_node
+        .wait_for_ingress_proofs_no_mining(vec![a_blk1_tx1.header.id], seconds_to_wait)
+        .await?;
+
+    // B: Mine B3
+
+    b_node.mine_block().await?;
+    network_height += 1;
+
+    b_node
+        .wait_until_height(network_height, seconds_to_wait)
+        .await?;
+
+    let b_blk3 = b_node.get_block_by_height(network_height).await?;
+
+    // ensure a_blk1_tx1 is included
+    assert_eq!(
+        b_blk3.data_ledgers[DataLedger::Submit].tx_ids,
+        vec![a_blk1_tx1.header.id]
+    );
+    assert_eq!(
+        b_blk3.data_ledgers[DataLedger::Publish].tx_ids, // should be promoted
+        vec![a_blk1_tx1.header.id]
+    );
+
+    assert!(b_blk3.data_ledgers[DataLedger::Publish].proofs.is_some());
+
+    // a_blk1_tx1 should have a new ingress proof (assert it's not the original from a_blk2)
+    assert!(b_blk3.data_ledgers[DataLedger::Publish]
+        .proofs
+        .clone()
+        .unwrap()
+        .ne(&IngressProofsList(vec![TxIngressProof {
+            proof: a_blk1_tx1_proof1.proof,
+            signature: a_blk1_tx1_proof1.signature
+        }])));
+
+    // now we gossip B3 back to A
+    // it shouldn't reorg, and should accept the block
+    // as well as overriding the ingress proof it has locally with the one from the block
+
+    b_node.send_block_to_peer(&a_node, &b_blk3).await?;
+
+    a_node
+        .wait_until_height(network_height, seconds_to_wait)
+        .await?;
+
+    assert_eq!(
+        a_node.get_block_by_height(network_height).await?,
+        b_node.get_block_by_height(network_height).await?
+    );
+
+    // assert that a_blk1_tx1 is no longer present in the mempool
+    // (nothing should be in the mempool)
+    let a_b_blk3_mempool_txs = a_node.get_best_mempool_tx(None).await?;
+    assert!(a_b_blk3_mempool_txs.submit_tx.is_empty());
+    assert!(a_b_blk3_mempool_txs.publish_tx.0.is_empty());
+    assert!(a_b_blk3_mempool_txs.publish_tx.1.is_empty());
+    assert!(a_b_blk3_mempool_txs.commitment_tx.is_empty());
+
+    // get a_blk1_tx1 from a, it should have b_blk3's ingress proof
+    let a_blk1_tx1_b_blk3_tx1 = a_node
+        .get_storage_tx_header_from_mempool(&a_blk1_tx1.header.id)
+        .await?;
+
+    assert_eq!(
+        a_blk1_tx1_b_blk3_tx1.ingress_proofs,
+        Some(
+            b_blk3.data_ledgers[DataLedger::Publish]
+                .proofs
+                .clone()
+                .unwrap()
+                .first()
+                .unwrap()
+                .clone()
+        )
+    );
+
+    // tada!
+
+    // gracefully shutdown nodes
+    tokio::join!(a_node.stop(), b_node.stop(), c_node.stop(),);
+    debug!("DONE!");
+    Ok(())
+}
+
+/// this test tests reorging in the context of confirmed commitment transactions
+/// goals:
+/// - ensure orphaned commitment transactions are returned to the mempool correctly
+/// Steps:
+/// create 2 nodes: A (genesis) B and C (C is currently unused)
```

Comment:
```suggestion
/// create 2 nodes: A (genesis) B
```

Simplify the test and remove all references of node C if it is not used